src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,157 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Parser {
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Parameterized constructor to create a Parser
+     * @param taskList the TaskList which has to be parsed
+     * @param ui the Ui of the agent on which messages have to be displayed
+     */
+    public Parser(TaskList taskList, Ui ui) {
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    /**
+     * Displays the current list of tasks
+     */
+    private void showList() {
+        ui.showTaskList(taskList);
+    }
+
+    /**
+     * Marks the Task at a specified index as done
+     * @param index the index of the Task to be marked as done
+     */
+    private void mark(int index) {
+        taskList.mark(index);
+        Task task = taskList.get(index);
+        ui.showMarkMessage(task, index);
+    }
+
+    /**
+     * Marks the Task at a specified index as undone
+     * @param index the index of the Task to be marked as undone
+     */
+    private void unmark(int index) {
+        taskList.unmark(index);
+        Task task = taskList.get(index);
+        ui.showUnmarkMessage(task, index);
+    }
+
+    /**
+     * Deletes the Task at a specified index
+     * @param index the index of the Task to be deleted
+     */
+    private void delete(int index) {
+        Task deletedTask = taskList.delete(index);
+        int len = taskList.size();
+        ui.showDeleteMessage(deletedTask, len);
+    }
```

Comment:
Perhaps can change to deleteCommand instead? Because if the method name is only delete then people might assume that the method is to delete something related to the Parser itself :)